tests/test_vmas.py

```diff
@@ -70,6 +72,135 @@ def test_multi_discrete_actions(scenario, num_envs=10, n_steps=10):
+        is not vmas.simulator.scenario.BaseScenario.process_action
+    ):
+        pytest.skip("Scenario uses a custom process_action method.")
+
+    random.seed(0)
+    for agent in env.world.agents:
+        agent.discrete_action_nvec = [
+            random.randint(2, 6) for _ in range(agent.action_size)
+        ]
+    env.action_space = env.get_action_space()
+
+    for _ in range(n_steps):
+        actions = env.get_random_actions()
+
+        # Check that generated actions are in the action space
+        for a_batch, s in zip(actions, env.action_space.spaces):
+            for a in a_batch:
+                assert a.numpy() in s
+
+        env.step(actions)
+
+        # Check that discrete action to continuous control mapping is correct.
+        for i_a, agent in enumerate(env.world.policy_agents):
+            for i, n in enumerate(agent.discrete_action_nvec):
+                a = actions[i_a][:, i]
+                u = agent.action.u[:, i]
+                U = agent.action.u_range_tensor[i]
+                k = agent.action.u_multiplier_tensor[i]
+                for aj, uj in zip(a, u):
+                    assert aj in range(
+                        n
+                    ), f"discrete action {aj} not in [0,{n-1}] (n={n}, U={U}, k={k})"
+                    if n % 2 != 0:
+                        assert (
+                            aj != 0 or uj == 0
+                        ), f"discrete action {aj} maps to control {uj} (n={n}), U={U}, k={k})"
+                        assert (aj < 1 or aj > n // 2) or torch.isclose(
+                            uj / k, (2 * U * (aj - 1)) / (n - 1) - U
+                        ), f"discrete action {aj} maps to control {uj} (n={n}, U={U}, k={k})"
+                        assert (aj <= n // 2) or torch.isclose(
+                            uj / k, 2 * U * (aj / (n - 1)) - U
+                        ), f"discrete action {aj} maps to control {uj} (n={n}), U={U}, k={k})"
+                    else:
+                        assert torch.isclose(
+                            uj / k, 2 * U * (aj / (n - 1)) - U
+                        ), f"discrete action {aj} maps to control {uj} (n={n}), U={U}, k={k})"
+
+
+@pytest.mark.parametrize("scenario", scenario_names())
+def test_discrete_action_nvec_discrete_to_multi(scenario, num_envs=10, n_steps=10):
+    kwargs = {
+        "scenario": scenario,
+        "num_envs": num_envs,
+        "seed": 0,
+        "continuous_actions": False,
+    }
+    env = make_env(**kwargs, multidiscrete_actions=False)
+    env_multi = make_env(**kwargs, multidiscrete_actions=True)
+    if (
+        type(env.scenario).process_action
+        is not vmas.simulator.scenario.BaseScenario.process_action
+    ):
+        pytest.skip("Scenario uses a custom process_action method.")
+
+    random.seed(0)
+    for agent, agent_multi in zip(
+        env.world.policy_agents, env_multi.world.policy_agents
+    ):
+        nvec = [random.randint(2, 6) for _ in range(agent.action_size)]
```

Comment:
Should we test with different action sizes? casue most scnearios have `action_size=2`. For the same reason, testing this on all scenarios is not really needed as the logic is in the environment. One scenario with a parametric `action_size` and `nvec` would be enough.